dlp/snippets/deid.py

```diff
@@ -1938,6 +1938,195 @@ def deidentify_table_with_multiple_crypto_hash(
 # [END dlp_deidentify_table_with_multiple_crypto_hash]
 
 
+# [START dlp_deidentify_table_fpe]
+import base64  # noqa: F811, E402, I100
+from typing import Dict, List, Union  # noqa: F811, E402, I100
+
+import google.cloud.dlp  # noqa: F811, E402
+
+
+def deidentify_table_with_fpe(
+    project: str,
+    table_data: Dict[str, Union[List[str], List[List[str]]]],
+    deid_content_list: List[str],
+    key_name: str = None,
+    wrapped_key: str = None,
+    alphabet: str = None,
+) -> None:
+    """Uses the Data Loss Prevention API to de-identify sensitive data in a
+      table while maintaining format.
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        table_data: Dictionary representing table data.
+        deid_content_list: A list of fields in table to de-identify.
+        key_name: The name of the Cloud KMS key used to encrypt ('wrap') the
+            AES-256 key. Example:
+            key_name = 'projects/YOUR_GCLOUD_PROJECT/locations/YOUR_LOCATION/
+            keyRings/YOUR_KEYRING_NAME/cryptoKeys/YOUR_KEY_NAME'
+        wrapped_key: The encrypted ('wrapped') AES-256 key to use. This key
+            should be encrypted using the Cloud KMS key specified by key_name.
+        alphabet: The set of characters to replace sensitive ones with. For
+            more information, see https://cloud.google.com/dlp/docs/reference/
+            rest/v2beta2/organizations.deidentifyTemplates#ffxcommonnativealphabet
+    """
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct the `table`. For more details on the table schema, please see
+    # https://cloud.google.com/dlp/docs/reference/rest/v2/ContentItem#Table
+    headers = [{"name": val} for val in table_data["header"]]
+    rows = []
+    for row in table_data["rows"]:
+        rows.append({"values": [{"string_value": cell_val} for cell_val in row]})
+
+    table = {"headers": headers, "rows": rows}
+
+    # Construct the `item` for table.
+    item = {"table": table}
+
+    # Specify fields to be de-identified.
+    deid_content_list = [{"name": _i} for _i in deid_content_list]
+
+    # The wrapped key is base64-encoded, but the library expects a binary
```

Comment:
Can we ask for 'bytes' here instead of a base64 string?

Reply:
You meant passing in the raw bytes, already decoded bytes?

Reply:
Yes. On the command line it makes sense to accept base64, but in code we should pass around bytes to make things simpler.

Reply:
I have updated for deid sample for now as per your suggestion, Please check if that is what is expected, If yes, I will update the same for reid sample too. Thank you!

Reply:
Yep, looks good. Thanks!